guides/assets/esa-initial-flow.txt

```diff
@@ -0,0 +1,40 @@
+sequenceDiagram
+    participant User
+    participant App
+    participant Application Route (ApplicationRouteMixin)
+    participant Index Route (AuthenticatedRouteMixin)
+    participant Login Route (UnauthenticatedRouteMixin)
+    participant Session Service
+    participant Torii Provider
+    participant GitHub
+    participant Token Exchanger
+    User->>App: Use https://git-tool.com
+    App->>Application Route (ApplicationRouteMixin): beforeModel()
+    Application Route (ApplicationRouteMixin)->>Application Route (ApplicationRouteMixin): subscribeToSessionEvents()
+    App->>Index Route (AuthenticatedRouteMixin): beoreModel()
+    Index Route (AuthenticatedRouteMixin)->>Session Service: isAuthenticated?
+    Session Service-->>Index Route (AuthenticatedRouteMixin): false
+    Index Route (AuthenticatedRouteMixin)->>Login Route (UnauthenticatedRouteMixin): transitionTo(authenticationRoute)
+    Login Route (UnauthenticatedRouteMixin)->>Login Route (UnauthenticatedRouteMixin): beforeModel()
+    Login Route (UnauthenticatedRouteMixin)->>Session Service: isAuthenticated?
+    Session Service-->>Login Route (UnauthenticatedRouteMixin): false
+    Login Route (UnauthenticatedRouteMixin)-->>App: render(loginTemplate)
+    App-->>User: Login screen
+    User->>App: Click "Login"
+    App->>Login Route (UnauthenticatedRouteMixin): login
+    Login Route (UnauthenticatedRouteMixin)->>Login Route (UnauthenticatedRouteMixin): controller.login()
```

Comment:
This isn't really what actually happens - the route calling the `login` method on the controller. Instead it might make sense to add the `Login` controller to the diagram and replace this and line 24 above with `App->>Login Controller: login action`?